test/recipe_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+)
+
+var testFilePaths = []string{
+	"ingress/single-cluster/",
+}
+
+func TestRecipe(t *testing.T) {
+	for _, fp := range testFilePaths {
+		fileNames, err := os.ReadDir(fp)
+		if err != nil {
+			t.Fatalf("Failed to open directory %s: %v", fp, err)
+		}
+		runRecipeTests(t, fp, fileNames)
+	}
+}
+
+// runRecipeTests iterates each recipe file, and run its test if it is a directory
+func runRecipeTests(t *testing.T, parentPath string, fileNames []fs.DirEntry) {
+	for _, fileName := range fileNames {
+		fileName := fileName
+		t.Run(fileName.Name(), func(t *testing.T) {
+			t.Parallel()
+			path := path.Join(parentPath, fileName.Name())
+			if err := validateDir(path); err != nil {
+				t.Skipf("Skipping test %q: validateDir(%q) = %v", path, path, err)
+			}
+			runRecipeTest(t, path)
+		})
+	}
+}
+
+// runRecipeTest runs the scripts for testing.
+func runRecipeTest(t *testing.T, recipeDir string) {
+	for _, file := range []string{"setup.sh", "run-test.sh", "cleanup.sh"} {
+		path := path.Join(recipeDir, file)
+		if _, err := os.Stat(path); err != nil {
+			t.Skipf("Skipping test %q: stat(%q) = %v", recipeDir, path, err)
```

Comment:
I'm not sure this is quite what we want to do.

We shouldn't partially the the scripts if not all of them exist.

Reply:
You should check the existence of ALL before you proceed to run the scripts.

Reply:
Updated.